Add --24hour flag to switch between 12-hour and 24-hour time formats

kingchenc opened the pull request:
📝 Usage Examples
Default 12-hour format:

ccusage blocks
# Output: Block Started: 1/15/2024, 2:30:45 PM (2h 15m ago)
24-hour format:

ccusage blocks --24hour
# Output: Block Started: 1/15/2024, 14:30:45 (2h 15m ago)
Live monitoring:

ccusage blocks --live --24hour
# Shows times like "Started: 14:30:45" instead of "Started: 02:30:45 PM"

Technical Details

Backwards Compatibility

✅ Flag is optional and defaults to 12-hour format
✅ All existing commands work unchanged
✅ Compatible with other flags like --json, --live, --breakdown

Implementation Approach

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
  - Added support for displaying times in 24-hour format across relevant commands and displays. Users can now enable this option via a new command-line argument.
- **Enhancements**
  - Improved date and time formatting consistency with new utilities, ensuring uniform output for both 12-hour and 24-hour formats.
- **Tests**
  - Introduced comprehensive tests for the new date and time formatting options.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ryoppippi commented:
I'm curios why we need this option. 24-hour is enough right?

kingchenc commented:
> I'm curios why we need this option. 24-hour is enough right?
> 
> Given the maintenance costs, we don't want to add more options without a fight, but I wonder how much demand there will be for this.

because im from europa and dont like am and pm and have 24 hours localy here?

kingchenc commented:
@ryoppippi 

Thanks for the feedback and for marking it as draft!

Totally agree – this touches broader i18n concerns like time, currency, and formatting preferences. I mainly wanted to open the discussion and show a working direction for 24h time as a user setting.

I wanted to keep it very simple — just an optional 24-hour time format toggle, since AM/PM is the default standard. The dollar currency display works fine for me.
Theoretically, country-specific currencies could be added, but that seems unnecessary for now.

Happy to adjust the implementation based on how you’d prefer this to be structured — e.g., a unified user config system or i18n module. Let me know if you'd like me to take a stab at that, or if you'd prefer to revisit it later.

Appreciate your thoughtful response!

ryoppippi commented:
@kingchenc Thanks


I'm thinking about this time option.

how about `--am-pm` for the option name and accepts boolean. it sounds good for me

kingchenc commented:
> @kingchenc Thanks
> 
> I'm thinking about this time option.
> 
> how about `--am-pm` for the option name and accepts boolean. it sounds good for me.
> 
> if you can change it, please do and make this PR ready!

Yes, of course, sounds good! I’ll do it tomorrow <3

kingchenc commented:
@ryoppippi 

I hope i understand you correctly now:

"how about --am-pm for the option name and accepts boolean. it sounds good for me." :)

Sooo = Change the --24hour flag to --am-pm with inverted logic.
If thats not correct, please tell me - Im a bit confused. :D

ryoppippi commented:
yeah it is!
we can discuss about how to name the option, but i've never seen number is used for option name!

kingchenc commented:
@ryoppippi 

 Sorry for the mess with the previous PR workflow - got a bit tangled up with git force-pushes earlier! 😅

  This implementation is now:
  - Clean and rebased on latest main
  - Updated to work with all recent changes
  - Uses `--am-pm` flag (no numbers in option names as requested)
  - Maintains 24-hour as default format

ryoppippi commented:
are you ready for my review? if so, press `ready for review`!!
I'm okay with this commit history!

kingchenc commented:
> are you ready for my review? if so, press `ready for review`!! I'm okay with this commit history!

Glad to hear that, awesome! <3

Done :)

ryoppippi commented:
@kingchenc I was busy today but I'll take a look at it

ryoppippi commented on src/_utils.ts:
```diff
@@ -303,6 +303,68 @@ export function formatCurrency(amount: number): string {
+ * @param options.compact - Whether to use compact date format (default: false)
+ * @returns Formatted date/time string
+ */
+export function formatDateTime(date: Date, useAmPm = false, options: {
+	includeDate?: boolean;
+	includeSeconds?: boolean;
+	compact?: boolean;
+} = {}): string {
+	const { includeDate = true, includeSeconds = false, compact = false } = options;
+
+	const baseOptions: Intl.DateTimeFormatOptions = {
+		hour: '2-digit',
+		minute: '2-digit',
+		hour12: useAmPm,
+	};
```
I think we need to add tests

ryoppippi commented on src/_utils.ts:
```diff
@@ -303,6 +303,68 @@ export function formatCurrency(amount: number): string {
+ * @param includeSeconds - Whether to include seconds (default: false)
+ * @returns Formatted time string
+ */
+export function formatTime(date: Date, useAmPm = false, includeSeconds = false): string {
+	const options: Intl.DateTimeFormatOptions = {
+		hour: '2-digit',
+		minute: '2-digit',
+		hour12: useAmPm,
+	};
+
+	if (includeSeconds) {
+		options.second = '2-digit';
+	}
+
+	return date.toLocaleTimeString(undefined, options);
```
test!
